### DISCUSSION TITLE
Is it possible to embed a resource in a VS Code Extension?

### DISCUSSION THREAD

**Discussion by mitsha-microsoft** [microsoft/vscode-discussions#1043] [Extension Development QnA] [answered]
I am authoring an internal VS Code Extension that needs to invoke a dotnet tool. I want to package the dll of the tool with the extension since the tool is not downloadable from any location. Is it possible that I can embed the dll inside the extension as a resource and then refer to it in the extension code?

**Comment by alefragnani** [comment, marked as answer]
Hi @mitsha-microsoft ,

Yes, you can embed a dll, exe, or any other tool/resource you need. The way you will use it inside your extension (other dependencies, path, etc), depends on the kind of the resource. 

Just make sure it will be located in a folder which is not _ignored_ by `vsce` while packaging it. 

Hope this helps

**Reply by mitsha-microsoft** [reply]
Hi @alefragnani,

The resource is an exe. I havent added it in the vscodeignore path. While debugging the extension, if I invoke the command that refers to this exe, it says not found. (From what I understand the current working directory is of VS Code's Executable). How do I get the extension's working directory?

**Reply by mitsha-microsoft** [reply]
Ah I found that I needed to use `context.extensionPath` [reference](https://code.visualstudio.com/api/references/vscode-api#ExtensionContext).

Thanks for the help! :)
